### PR TITLE
Adityaahuja/btech datafetch

### PR DESCRIPTION
**Pull request by adityaahuja7**
- Fixed styling for student tables. (Fixed method passing logic by caching the method on the component and just passing data from the page)
- Fully functional data fetching from google sheets.
- Have build tested in my local environment. 
- Note: Please add the env file before testing.

### CODE REVIEW

File: app/people/btech/page.js [hunk 1]
```diff
@@ -68,16 +21,21 @@ const rows = [
...
   [18, 2020276, 'ADITYA AHUJA', 2020],
   [19, 2020277, 'BHAGESH VERMA', 2020],
-  [20, 202315, 'DIVYAM SINGH', 2020],
+  [20, 2020315, 'DIVYAM SINGH', 2020],
 ];
 
-export default function BtechStudents() {
+export default async function BtechStudents() {
+  const spreadsheetId = '1cYPbswNr4ZvvQhzkfIOaM9Q75cSDP1CSl_JELvmA4iQ';
+  const range = 'B-Tech!A2:D1000';
```
Comment (aryanteng) [inline app/people/btech/page.js:29]: i mentioned on the group, you don't need to give cells in range, just give B-Tech and entire sheet will come, if we give ranges, when admins add data, it won't reflect.

File: components/common/BtechDataTable.js [hunk 2]
```diff
@@ -1,21 +1,88 @@
+'use client';
 import React, { useCallback } from 'react';
 import MUIDataTable from 'mui-datatables';
 import { createTheme, ThemeProvider } from '@mui/material/styles';
+import LinkedIn from '@mui/icons-material/LinkedIn';
+
+function DataTable({ data, initialRows }) {
```
Comment (aryanteng) [inline components/common/BtechDataTable.js:7]: why is same code being repeated in this and mtech data table? create another component and name it DataTable and then add it to BtechDataTable and MtechDataTable, please don't repeat code.

File: components/common/MtechDataTable.js [hunk 3]
```diff
@@ -0,0 +1,212 @@
+'use client';
+import React, { useCallback } from 'react';
+import MUIDataTable from 'mui-datatables';
+import { createTheme, ThemeProvider } from '@mui/material/styles';
+import LinkedIn from '@mui/icons-material/LinkedIn';
+
+function DataTable({ data, initialRows }) {
```
Comment (aryanteng) [inline components/common/MtechDataTable.js:7]: change the name to MtechDataTable

**Review by aryanteng — CHANGES_REQUESTED** [review]

**Comment by aryanteng** [comment]
you are confused between btech and mtech I guess, in the deployment I saw mtech data from spreadsheets but btech data was same dummy data

**Comment by adityaahuja7** [comment]
Added wrapper components please check @aryanteng.

**Comment by aryanteng** [comment]
Screenshot 2024-03-09 at 2 02 34 PM
first row has unwanted data @adityaahuja7

File: components/people/btech/BtechDataTable.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      },
+    ];
+  });
+
+  return (
+    <DataTable
+      data={data}
+      columns={getColumns()}
+      initialRows={initialRows}
+      options={{
```
Comment (aryanteng) [inline components/people/btech/BtechDataTable.js:61]: no need for this, it is already in the DataTable component

File: components/people/mtech/MtechDataTable.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+      },
+    ];
+  });
+
+  return (
+    <DataTable
+      data={data}
+      columns={getColumns()}
+      initialRows={initialRows}
+      options={{
```
Comment (aryanteng) [inline components/people/mtech/MtechDataTable.js:78]: no need

**Review by aryanteng — CHANGES_REQUESTED** [review]

File: components/people/btech/BtechDataTable.js (hunk 4 above)
Comment (adityaahuja7) [reply components/people/btech/BtechDataTable.js:61]: You mean the options props right?

Comment (aryanteng) [reply components/people/btech/BtechDataTable.js:61]: yes

**Review by aryanteng — APPROVED** [review]

**Pull request merged**